include/aspect/time_stepping/repeat_on_nonlinear_fail.h

```diff
@@ -0,0 +1,102 @@
+/*
+  Copyright (C) 2018 - 2023 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_time_stepping_repeat_on_nonlinear_fail_h
+#define _aspect_time_stepping_repeat_on_nonlinear_fail_h
+
+#include <aspect/time_stepping/interface.h>
+
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements a time stepping plugin to repeat a time step if the
+     * nonlinear solver failed to converge in the specified number of iterations.
+     * This class is automatically enabled if the user specifies that he/she wants
+     * to do this.
```

Comment:
Can you add a sentence why the result for the next try would be different from the previous one? Currently it reads as if you may get a different result if you only reran an exactly identical problem.